modules/standard/Map.chpl

```diff
@@ -105,20 +113,32 @@ module Map {
       :arg keyType: The type of the keys of this map.
       :arg valType: The type of the values of this map.
       :arg parSafe: If `true`, this map will use parallel safe operations.
+      :arg resizeThreshold: Fractional value that specifies how full this map
+                            can be before requesting additional memory.
     */
-    proc init(type keyType, type valType, param parSafe=false) {
+    proc init(type keyType, type valType, param parSafe=false,
+              resizeThreshold: real = 0.5) {
       _checkKeyAndValType(keyType, valType);
       this.keyType = keyType;
       this.valType = valType;
       this.parSafe = parSafe;
+      assert((resizeThreshold > 0 && resizeThreshold < 1),
+             "'resizeThreshold' must be between 0 and 1");
+      this.resizeThreshold = resizeThreshold;
+      table = new chpl__hashtable(keyType, valType, resizeThreshold);
     }
 
-    proc init(type keyType, type valType, param parSafe=false)
+    proc init(type keyType, type valType, param parSafe=false,
+              resizeThreshold: real = 0.5)
     where isNonNilableClass(valType) {
       _checkKeyAndValType(keyType, valType);
       this.keyType = keyType;
       this.valType = valType;
       this.parSafe = parSafe;
+      assert((resizeThreshold > 0 && resizeThreshold < 1),
+             "'resizeThreshold' must be between 0 and 1");
```

Comment:
I think asserts are good. But a more typical module code can look like this:

```chapel
if boundsChecking then
  if resizeThreshold > 0 && resizeThreshold < 1 then
    boundsCheckHalt("'resizeThreshold' .... ");
```

`boundsChecking` is a param flag. it is on by default, but `--fast` and `--no-checks` turn it off.
`boundsCheckHalt` is something I have seen in the rest of the `Map` code, and just a wrapper around `halt`.